proton-c/src/tests/fuzz/fuzz-sniff-header.c

```diff
@@ -0,0 +1,15 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "core/autodetect.h"
+
+#ifdef __cplusplus
```

Comment:
There is a project specific cmake option BUILD_WITH_CXX which will build .c files with C++ compiler. See e.g. https://github.com/apache/qpid-proton/blob/9ec97b51c1fed9a77e5c3dc537e301fb7ae3c5fe/proton-c/src/tests/CMakeLists.txt#L30 Thank you very much for review.